dota_profanity_checker/game.py

```diff
@@ -0,0 +1,89 @@
+from pydantic import BaseModel, validator
+from typing import List
+
+from dota_profanity_checker.chat import Chat
+from dota_profanity_checker.player import Player
+from dota_profanity_checker.profanity import ProfanityChecker
+
+
+class Game(BaseModel):
+    """Class representing a single dota game."""
+    match_id: int
+    players: List[Player]
+    chat: List[Chat] = None
+    _chat_assigned: bool = False
+    _profanity_checker = ProfanityChecker()
+
+    class Config:
+        underscore_attrs_are_private = True
+        validate_assignment = True
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @validator("chat")
+    def set_chat(cls, chat):
```

Comment:
It technically does both, it validates if chat is present and sets a default value (an empty list) if it isn't. I'll be honest, this @validator decorator is a quirk of the pydantic.BaseModel that I don't really understand. In a regular class this would be much simpler to do. I think I dove in a little quick with using pydantic! Have changed the name for now however.